src/models/converters/balances.rs

```diff
@@ -11,7 +11,13 @@ impl BalanceDto {
             .token_address
             .as_ref()
             .map(|_| TokenType::Erc20)
-            .unwrap_or(TokenType::Ether);
+            .unwrap_or(TokenType::NativeToken);
+
+        let logo_uri = if token_type == TokenType::NativeToken {
+            Some(native_coin.logo_url.to_string())
+        } else {
+            self.token.as_ref().map(|it| it.logo_uri.to_string())
+        };
```

Comment:
What's your opinion on removing the `native_coin` entirely from this function?

This is related to the balance of a specific token right? Wouldn't providing the `native_coin` as a fallback be incorrect? 🤔

Reply:
When the core service don't return an address we assume that it is the native token. And as the clients would like to have the `logo_uri` required for the native coind this is the correct way to provide it